### PR TITLE
util: Fix Coverity issues in NS functionality

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by shefty** [comment]
thanks

File: prov/util/src/util_ns.c [hunk 1]
```diff
@@ -285,15 +285,16 @@ static void *util_ns_name_server_func(void *args)
...
-				break;
-			ofi_close_socket(listenfd);
-			listenfd = INVALID_SOCKET;
+			if (setsockopt(listenfd, SOL_SOCKET,
+				       SO_REUSEADDR, &n, sizeof(n)) ||
+			    (bind(listenfd, p->ai_addr, p->ai_addrlen))) {
+				ofi_close_socket(listenfd);
+				listenfd = INVALID_SOCKET;
+			}
 		}
```
Comment (j-xiong) [inline prov/util/src/util_ns.c]: The change breaks the name server. The loop would continue even if bind succeed and in the end we got invalid listenfd unless the succeeded bind happens to the last one. 

Massive fabtests failures have been observed over psm and psm2 with this patch.

I suggest we revert this PR and take a differet approach for the Coverity issue. 

@shefty @gladkovdmitry17

Comment (j-xiong) [inline prov/util/src/util_ns.c]: Also the setsockopt() call is non fatal. Any error returned from that should be ignored.

Comment (shefty) [inline prov/util/src/util_ns.c]: Are the failures only associated with this patch or the full ns patch?

Comment (j-xiong) [inline prov/util/src/util_ns.c]: Just this patch. Specifically those line above my comments.

Comment (shefty) [inline prov/util/src/util_ns.c]: Reverted by #3111

Comment (gladkovdmitry17) [inline prov/util/src/util_ns.c]: Sorry for my bad. I checked on my system and it was OK. I'll return previous logic of loop

**Pull request merged**